### PR TITLE
[DO NOT MERGE] C100-Rebuild Release 4.0 Changes - WIP

### PR DESCRIPTION
**Pull request by channa-patil**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by yogendraU — APPROVED** [review]

File: definitions/private-law/json/AuthorisationCaseEvent/AuthorisationCaseEvent.json [hunk 1]
```diff
@@ -105,5 +105,19 @@
...
+    "CaseEventID": "citizen-case-submit-with-hwf",
+    "UserRole": "citizen",
+    "CRUD": "CRU"
+  },
+  {
+    "LiveFrom": "01/08/2021",
+    "CaseTypeID": "PRLAPPS",
+    "CaseEventID": "citizen-case-submit-with-hwf",
+    "UserRole": "caseworker-privatelaw-courtadmin",
+    "CRUD": "R"
```
Comment (AlokDatta) [inline definitions/private-law/json/AuthorisationCaseEvent/AuthorisationCaseEvent.json:121]: can we do it with access control. we will have reduced number of code. also authorisation should be granted to judges, la, system user, system update in order to view it in history.

File: definitions/private-law/json/AuthorisationCaseField/C100Rebuild/AuthorisationCaseField.json [hunk 2]
```diff
@@ -236,5 +236,19 @@
...
+    "CaseFieldID": "noOfDaysRemainingToSubmitCase",
+    "UserRole": "citizen",
+    "CRUD": "CRU"
+  },
+  {
+    "LiveFrom": "01/01/2023",
+    "CaseTypeID": "PRLAPPS",
+    "CaseFieldID": "noOfDaysRemainingToSubmitCase",
+    "UserRole": "caseworker-privatelaw-courtadmin",
+    "CRUD": "R"
```
Comment (AlokDatta) [inline definitions/private-law/json/AuthorisationCaseField/C100Rebuild/AuthorisationCaseField.json:252]: can we do it with access control. we will have reduced number of code. also authorisation should be granted to judges, la, system user, system update in order to view it in history.

File: definitions/private-law/json/CaseEvent.json [hunk 3]
```diff
@@ -1721,5 +1721,18 @@
     "SecurityClassification": "Public",
     "ShowEventNotes": "N",
     "EventEnablingCondition": "caseTypeOfApplication=\"FL401\""
+  },
+  {
+    "LiveFrom": "01/07/2021",
+    "CaseTypeID": "PRLAPPS",
+    "ID": "citizen-case-submit-with-hwf",
```
Comment (AlokDatta) [inline definitions/private-law/json/CaseEvent.json:1728]: can we align this id to the existing ones like citizenCaseSubmitWithHWF, if possible please.

**Review by AlokDatta — CHANGES_REQUESTED** [review]

File: definitions/private-law/json/CaseEvent.json (hunk 3 above)
Comment (channa-patil) [reply definitions/private-law/json/CaseEvent.json:1728]: Done

File: definitions/private-law/json/AuthorisationCaseField/C100Rebuild/AuthorisationCaseField.json (hunk 2 above)
Comment (channa-patil) [reply definitions/private-law/json/AuthorisationCaseField/C100Rebuild/AuthorisationCaseField.json:252]: Done

File: definitions/private-law/json/AuthorisationCaseEvent/AuthorisationCaseEvent.json (hunk 1 above)
Comment (channa-patil) [reply definitions/private-law/json/AuthorisationCaseEvent/AuthorisationCaseEvent.json]: Done

**Pull request merged**